### PR TITLE
Threads Page implementation

### PR DESCRIPTION
**Pull request by pauleke65**

### CODE REVIEW

File: lib/models/user_post.dart [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+enum ChannelType { private, public, personal }
```
Comment (Quwaysim) [inline lib/models/user_post.dart:4]: move enum to lib/utilities/enums.dart

File: lib/ui/shared/bottom_sheets/custom_user_bottomsheet/custom_user_bottom_sheet_view.dart [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                        Divider(),
+                        CustomProfileTile(
+                            title: 'Display Name', subtitle: 'pauleke65'),
+                        Divider(),
+                        ListTile(
+                          title: CustomText(
+                              text: 'Status', fontWeight: FontWeight.w300),
+                          subtitle: Align(
+                              alignment: Alignment.centerLeft,
+                              child: Icon(Icons.looks_5, color: Colors.blue)),
```
Comment (Quwaysim) [inline lib/ui/shared/bottom_sheets/custom_user_bottomsheet/custom_user_bottom_sheet_view.dart:75]: extract colors to the AppColors class

File: lib/ui/shared/bottom_sheets/custom_user_bottomsheet/widgets/custom_button.dart [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import 'package:flutter/material.dart';
+import 'package:hng/ui/shared/colors.dart';
+
+enum _CustomButtonType { normal, icon }
```
Comment (Quwaysim) [inline lib/ui/shared/bottom_sheets/custom_user_bottomsheet/widgets/custom_button.dart:4]: move to enums.dart file

File: lib/ui/shared/bottom_sheets/custom_user_bottomsheet/widgets/custom_button.dart [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return _customButtonType == _CustomButtonType.normal
+        ? MaterialButton(
+            onPressed: onPressed,
+            padding: EdgeInsets.all(MediaQuery.of(context).size.height * 0.02),
+            shape: Border.all(color: AppColors.greyishColor),
+            child: Text(
+              " $text",
+              style: TextStyle(color: AppColors.greyishColor),
```
Comment (Quwaysim) [inline lib/ui/shared/bottom_sheets/custom_user_bottomsheet/widgets/custom_button.dart:31]: Use one of the const TextStyles or extract yours to the file

File: lib/ui/view/threads/threads_view.dart [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+          child: SingleChildScrollView(
+            physics: const AlwaysScrollableScrollPhysics(),
+            child: Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              mainAxisSize: MainAxisSize.max,
+              children: [
+                Padding(
+                  padding: EdgeInsets.symmetric(horizontal: 15, vertical: 10),
+                  child: CustomText(
+                    text: 'No new replies',
```
Comment (Quwaysim) [inline lib/ui/view/threads/threads_view.dart:39]: use one of the const styles or create yours in styles.dart

File: lib/ui/view/threads/widgets/thread_card.dart [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+                SizedBox(width: 5),
+                CustomText(
+                  text: userPost!.channelName ?? '',
+                  fontSize: 14,
+                  fontWeight: FontWeight.bold,
+                ),
+              ],
+            ),
+            subtitle: CustomText(
+              text: 'You, Jane and 2 others',
```
Comment (Quwaysim) [inline lib/ui/view/threads/widgets/thread_card.dart:68]: styles.dart

File: lib/ui/view/threads/widgets/thread_post.dart [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+                          ),
+                          Icon(
+                            Icons.looks_6,
+                            color: Colors.blue,
+                            size: 18,
+                          ),
+                          SizedBox(width: 2),
+                          CustomText(
+                            text: "${userThreadPost.lastSeen}",
+                            fontSize: 12,
```
Comment (Quwaysim) [inline lib/ui/view/threads/widgets/thread_post.dart:58]: styles here too

**Review by Quwaysim — CHANGES_REQUESTED** [review]
please make changes...

GREAT work!!

**Review by pauleke65 — COMMENTED** [review]
Changes made

**Review by pauleke65 — COMMENTED** [review]
changes made

**Review by Quwaysim — APPROVED** [review]
LG

**Comment by mariamadebolahamzat** [comment]
Resolve conflicts

**Review by mariamadebolahamzat — APPROVED** [review]
All changes made

**Review by Quwaysim — APPROVED** [review]

**Pull request merged**